catkin_home/src/main_engine/src/Action.py

```diff
@@ -0,0 +1,71 @@
+import rospy
+import actionlib
+import time
+#from object_detection.msg import RecognizeObjectAction, RecognizeObjectGoal
+# TODO: Import all the messages related to actions I guess ?
+'''
+All Goals, and Actions must be imported here from he corresponding packages
+'''
+
+
+class Action(object):
+    def __init__(self, id, priority, action_client_binded, category, args):
+        self.id = id
+        self.priority = int(priority)
+        self.args = args
+        self.category = int(category)
+        #Possible States Are: PENDING, ACTIVE, RECALLED, REJECTED, PREEMPTED, ABORTED, SUCCEEDED, LOST.
+        self.feedback = ["PENDING", "ACTIVE", "RECALLED", "REJECTED", "PREEMPTED", "ABORTED", "SUCCEEDED", "LOST"]
```

Comment:
Is the `RECALLED` state needed?

Reply:
Those are the states defined in ROS for Feedback:
https://docs.ros.org/api/actionlib/html/classactionlib_1_1simple__action__client_1_1SimpleActionClient.html#a1496dbc011f48451f4ea98e1ad2f8cd9